### PR TITLE
Add dispatching file creation and deletion event

### PR DESCRIPTION
**Pull request by tkawaji**
Hello,

In this PR, I want to hook creation and deletion of files for uploading to other service (Amason S3, etc).
So I added functions which dispatch "Proffer.afterCreateImage" and "Proffer.beforeDeleteFolder" events to beforeSave and afterDelete.

Thank you.

### CODE REVIEW

File: src/Model/Behavior/ProfferBehavior.php [hunk 1]
```diff
@@ -101,7 +105,14 @@ public function beforeSave(Event $event, Entity $entity, ArrayObject $options, P
                             $imageTransform = new ImageTransform($this->_table, $path);
                         }
 
-                        $imageTransform->processThumbnails($settings);
+                        $thumbnailPaths = $imageTransform->processThumbnails($settings);
```
Comment (davidyell) [inline src/Model/Behavior/ProfferBehavior.php]: I'm not sure firing events in a loop is wise here.

**Comment by davidyell** [comment]
Looks good, any chance you can update the documentation as well to add the new events?

File: src/Model/Behavior/ProfferBehavior.php (hunk 1 above)
Comment (tkawaji) [inline src/Model/Behavior/ProfferBehavior.php]: Thank you for your reply!

I'm not sure, too.
Do you think it's better to fix to pass array?

**Comment by tkawaji** [comment]
Document list which I will update is below. That's ok?

\<edit>
docs/customisation.md
(add customization description)

docs/examples.md
(add link to /UploadAndDeleteImageListener.md)

\<new>
docs/examples/UploadAndDeleteImageListener.md

**Comment by davidyell** [comment]
Yes please, that would be great, then I can merge this.

**Comment by tkawaji** [comment]
How about above inline comment?
But I think that for EventListener it's better to be able to treat all files at once because the cost for initializing class treating services (such as Aws Class) cannot be ignored.
Do you know that EventListener instance is same when hooking several events? (So I can initialize service object only once and set it into instance variable.)

**Comment by tkawaji** [comment]
I tested that.

In bootstrap.php
```php
Cake\Event\EventManager::instance()->on(new \App\Event\UploadAndDeleteImageListener());
```

listener instance is same across several events.
So I think either is ok about above problem.

File: src/Model/Behavior/ProfferBehavior.php (hunk 1 above)
Comment (davidyell) [inline src/Model/Behavior/ProfferBehavior.php]: Ahh ok, I didn't know that. Using the same instance sounds good to me.

**Comment by davidyell** [comment]
Ace, thanks for all the work on this!

**Pull request merged**